repair/row_level.cc

```diff
@@ -519,12 +529,28 @@ class repair_writer {
         t.stream_in_progress());
     }
 
+    future<> write_partition_end(unsigned node_idx) {
+        if (_stats[node_idx].nr_parttion_start == _stats[node_idx].nr_parttion_end + 1) {
+            return _mq[node_idx]->push_eventually(mutation_fragment(partition_end())).then([this, node_idx] {
+                _stats[node_idx].nr_parttion_end++;
+            });
```

Comment:
Again this looks race prone (since repair aborts are asynchronous to this code), nr_parttion_end increments can be delayed.

Reply:
Why not use seastar::queue::abort()? It is designed for asynchronous operation, and can be called when the producer is blocking in push_eventually().

Reply:
Abort is actually synchronous, it checks if the abort flag is set in each repair steps inside the main repair loop.

Reply:
I don't understand either the problem or the solution.

Reply:
If it is syncronous, why is there a problem? Why did we call write_partition_end()? What's the call site from abort's persepective?

Maybe abort shouldn't write the partition end if no partition is open?